### ISSUE TITLE
[Feature Request] Add `destroyInactiveTabPanel` prop for Tabs component

### ISSUE THREAD

**Issue by u3u** [nextui-org/nextui#1562]
### Is your feature request related to a problem? Please describe.

In some cases, I hope that when switching tabs, the previous TabPanel will not be destroyed in order to preserve their state and avoid re-rendering.

### Describe the solution you'd like

We can add a `destroyInactiveTabPanel` prop with a default value of `false`.
The inactive TabPanel can be hidden using `display: none`.
If the user wants to destroy the previous TabPanel every time they switch tabs, they can simply add it to the Tabs component.

### Describe alternatives you've considered

Now can also achieve the same effect by yourself, but each time you need to write additional code or encapsulate it again.

```tsx
const [selectedKey, setSelectedKey] = useState<TabKey>('basic');
const [canRenderTabs, setCanRenderTabs] = useState([selectedKey]);

const setSelectedTab = (key: TabKey) => {
  if (!canRenderTabs.includes(key)) setCanRenderTabs([...canRenderTabs, key]);
  setSelectedKey(key);
};

{canRenderTabs.map((key) => {
  const Content = tabsMap[key];
  const isSelected = key === selectedKey;

  return (
    <div className={twMerge(styles.panel, !isSelected && 'hidden')} key={key}>
      <Content />
    </div>
  );
})}
```

**Comment by Softwaremanufaktur** [comment]
when creating a brand new project with nextui 2.4.0 the property "destroyInactiveTabPanel" is not recognized (yet) :-/
```
// package.json
...
"dependencies": {
		"@nextui-org/react": "^2.4.0",
...
```
![grafik](https://github.com/nextui-org/nextui/assets/47534220/9bcbde6e-6719-4a6a-a67b-454a9692f9c6)

**Comment by wingkwong** [comment]
@Softwaremanufaktur `destroyInactiveTabPanel` should be a prop for `Tabs` instead of `Tab`. The doc is incorrect. I'll update it.

**Comment by Neuro8** [comment]
Hi, I updated NextUI to 2.4.0 but the prop is not changing anything, my tab and the component within it keep being re rendered each time we switch tabs. Any idea why it could be the case?

I made sure to put the prop in Tabs as indicated above.

Thanks

**Comment by wingkwong** [comment]
@Neuro8 can u setup a minimal reproducible stackblitz project for us to investigate?

**Comment by drewdearing** [comment]
any update here? Setting destroyInactiveTabPanel to false does not prevent tab destruction.

**Comment by wingkwong** [comment]
@drewdearing the fix has been deployed to 2.4.2. Are you using the latest version?

**Comment by drewdearing** [comment]
> @drewdearing the fix has been deployed to 2.4.2. Are you using the latest version?

Oh I didn't realize there was a fix based on this thread. I'll give it a try.
